tiledb/sm/c_api/tiledb.cc

```diff
@@ -2930,6 +2930,11 @@ int32_t tiledb_array_encryption_type(
   auto storage_manager = ctx->storage_manager();
   auto uri = tiledb::sm::URI(array_uri);
 
+  if (uri.is_tiledb()) {
+    throw std::invalid_argument(
+        "Getting the encryption type of remote arrays is not supported.");
+  }
+
```

Comment:
How about we return no encryption instead of failing?

Reply:
@KiterLuc opinion on this?

Reply:
I believe we should keep logic out of the C-api. Can this be moved into the C++ side?

Reply:
I can't think of a better place to have this logic. In the past this could fit in a `StorageManager::get_array_encryption_type` function, but we want to stop using `StorageManager`.

Reply:
Oh there was already such method, great.